build/go.bzl

```diff
@@ -0,0 +1,50 @@
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A wrapper for go_library() which adds .gofiles, .sourcefiles filegroup targets.
+#
+# Usage:
+#   replace lines like:
+#     load("@io_bazel_rule_go//go:def.bzl", "go_library")
+#   with:
+#     load("//build:go.bzl", "go_library")
+#
+#   Now a //foo:go_default_library also comes with //foo:go_default_library.sourcefiles
+
+load("@io_bazel_rules_go//go:def.bzl",
+     "go_test",
+     "go_binary",
+     "go_prefix",
+     _go_library="go_library",
+)
+
+def go_library(name, srcs=None, deps=None, **kw):
+  """Wrap go_library() to also produce a .gofiles and .sourcefiles targets."""
+  # Create the actual library rule
+  _go_library(name=name, srcs=srcs, deps=deps, **kw)
+  # Now also create the .gofiles and .sourcefiles rules
+  srcs = srcs or []
+  deps = deps or []
+  # All go files in this package
+  native.filegroup(
+    name = name+".gofiles",
```

Comment:
what's the use of `.gofiles`? I don't see anything using it in the other PR.

Reply:
Doesn't harm anything and may be useful in the future (make kazel unnecessary, perhaps). Would you prefer I remove it until it is used?